spec/api/json_rpc_spec.rb

```diff
@@ -179,5 +180,30 @@ def wait_for_expect(retry_count = 20, sleep_duration = 0.5)
         expect(last_json_response).to include(expected_error_response)
       end
     end
+
+    context "when an internal error triggers the sinatra error handler" do
+      before(:each) do
+        allow_any_instance_of(Msf::RPC::JSON::Dispatcher).to receive(:process) do
+          raise Exception, "Something imploded"
+        end
+      end
+
+      it 'returns the error results' do
+        create_job
+
+        expect(last_response).to be_server_error
+        expected_error_response = {
+          "error" => {
+            "code" => -32000,
+            "data" => {
+              "backtrace" => include(a_kind_of(String))
+            },
+            "message" => "Application server error: Something imploded"
```

Comment:
Does this check that the error message is _exactly_ this? that can lead to kind of fragile tests, I don't think the tests should break if someone decided to make a minor change to the message structure

Reply:
A change to the `Application server error: ` part of the message would be caused by an alteration or replacement of the  `APPLICATION_SERVER_ERROR` class, which should be reflected in the test. The `Something imploded` ensures we get the name of the exception exactly. I get what you mean it is brittle, but for what the test checks I think it's good enough